analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.ExceptionCandidate.cs

```diff
@@ -0,0 +1,549 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.UnitTest.SymbolicExecution.Roslyn
+{
+    public partial class RoslynSymbolicExecutionTest
+    {
+        [TestMethod]
+        public void ExceptionCandidate_ArrayElementReference()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    var x = c[42];
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code, ", char[] c").Validator;
+            validator.ValidateContainsOperation(OperationKindEx.ArrayElementReference);
+            validator.ValidateTagOrder("BeforeTry", "InTry", "InCatch", "AfterCatch");
+            validator.TagStates("InCatch").Should().HaveCount(1).And.ContainSingle(x => HasUnknownException(x));
+            validator.ExitStates.Should().ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_ObjectCreation()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    var x = new String(c, 0, 0); // Can throw ArgumentNullException
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code, ", char[] c").Validator;
+            validator.ValidateContainsOperation(OperationKindEx.ObjectCreation);
+            validator.ValidateTagOrder("BeforeTry", "InTry", "InCatch", "AfterCatch");
+            validator.TagStates("InCatch").Should().ContainSingle(x => HasUnknownException(x));
+            validator.ExitStates.Should().ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_ConversionOperation_Explicit()
```

Comment:
To align with the `_Widening` suffix below
```suggestion
        public void ExceptionCandidate_ConversionOperation_Explicit_Narrowing()
```

Reply:
To avoid change -> build -> rebase -> build cycle I will do both changes in https://github.com/SonarSource/sonar-dotnet/pull/5783